website/docs/API.md

```diff
@@ -515,22 +517,94 @@ Defined as:
 ```jsx
 function waitFor<T>(
   expectation: () => T,
-  { timeout: number = 4500, interval: number = 50 }
+  { timeout: number = 1000, interval: number = 50 }
 ): Promise<T> {}
 ```
 
-Waits for non-deterministic periods of time until your element appears or times out. `waitFor` periodically calls `expectation` every `interval` milliseconds to determine whether the element appeared or not.
+Should be used when testing asynchronous code or code with timers. `waitFor` allows to wait for some expectation (e.g. the presence of an element on the screen). The `expectation` param is a callback that should throw until what you're waiting for has realized.
+
+For instance, this is how we'd wait for an element to appear:
+
+```tsx
+await waitFor(() => expect(screen.getByText('Banana ready').toBeTruthy()))
+```
+
+What `waitFor` does is executing `expectation()` at intervals of duration equal to the optional `interval` param (default is 50ms) until it doesn't throw anymore. It will throw if it reaches its timeout, defined by the optional `timeout` param (default is 1000ms). If `expectation()` does not throw then `waitFor` will resolve right away.
```

Comment:
This is a bit of simplification, as under node `setTimeout`/`setInterval` are not guaranteeing exact amount of timeout/interval but rather "at least" amount. They might be run later, e.g. due to JS thread being busy by some longer computation. I don't think we need to explain the whole story here, but just tweak the wording a bit:
``waitFor` periodically calls `expectation` approximately every `interval` milliseconds`

Reply:
You're right I hadn't thought of this. However that's never really been an issue for me, I wonder if it's worth mentioning in the doc. I fear that the term `approximately` may just be the source of more confusion as it's really not cleat as to why it would be approximated, is there value in users knowing this ? wdyt ?

Reply:
I agree with Pierre about "approximately", maybe we could add a little note at the end saying: "Under the hood, waitFor uses setTimeout and setInterval so the `interval` and `timeout` values are "at least" amounts. They might be run later if the JS thread is busy. More on that [here](https://www.youtube.com/watch?v=8aGhZQkoFbQ)" (I pointed to one ressource but have no strong preference)

Reply:
Maybe this should be detailed in a section about waitFor but I wouldn't add so much precision in this part of the doc, wdyt ?

Reply:
Agreed